pkg/k8s/k8s_statefulset.go

```diff
@@ -0,0 +1,185 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	appv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// getStatefulSet retrieves a statefulSet from the given namespace and logs any errors.
+func getStatefulSet(namespace, name string) (*appv1.StatefulSet, error) {
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
+	if !IsInitialized() {
+		return nil, fmt.Errorf("knuu is not initialized")
+	}
+	statefulset, err := Clientset().AppsV1().StatefulSets(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get statefulSet %s: %w", name, err)
+	}
+
+	return statefulset, nil
+}
+
+// DeployStatefulSet creates a new statefulSet in the given namespace if it doesn't already exist.
+func DeployStatefulSet(statefulSetConfig StatefulSetConfig, init bool) (*appv1.StatefulSet, error) {
```

Comment:
maybe we can call it `CreateStatefulSet`

Reply:
just to match the references to `create` within the function

Reply:
Yeah, the `k8s` package is a mess currently. The naming could be more consistent, but I created an issue for that. See https://github.com/celestiaorg/knuu/issues/108 and https://github.com/celestiaorg/knuu/issues/111.